src/types/courses/assignment.ts

```diff
@@ -16,3 +16,21 @@ export interface AssignmentConfig {
   userId: string;
   previewFileName: string;
 }
+
+export interface ISubmissionContent {
+  _type: string;
+}
+
+export interface IProgrammingSubmissionContent extends ISubmissionContent {
+  [key: string]: string;
+}
+
+export interface ITextSubmissionContent extends ISubmissionContent {
+  text: string;
+}
+export interface IUrlSubmission extends ISubmissionContent {
+  url: string;
+  service: string;
+}
+
+export type ISubmissionTypeContent = IProgrammingSubmissionContent | ITextSubmissionContent | IUrlSubmission | string;
```

Comment:
`ISubmissionTypeContent` is not required as the interface is already present `ISubmissionContent`